projects/ngx-d3timeline/src/lib/store/actions.ts

```diff
@@ -0,0 +1,40 @@
+import { TimelineEvent } from '../timeline-event';
+import { Orientation } from '../orientation';
+
+export interface Action {
+  type: string;
+  payload?: any;
+}
+
+export enum ActionTypes {
+  DataChanged = 'Data Changed',
+  OrientationChanged = 'Orientation Changed',
+  ViewChanged = 'View Changed',
+  ZoomEvent = 'Zoom Event'
+}
+
+export class DataChangedAction implements Action {
+  readonly type = ActionTypes.DataChanged;
+  constructor(public payload: TimelineEvent[]) {}
+}
+
+export class OrientationChangedAction implements Action {
+  readonly type = ActionTypes.OrientationChanged;
+  constructor(public payload: Orientation) {}
+}
+
+export class ViewChangedAction implements Action {
+  readonly type = ActionTypes.ViewChanged;
+  constructor(public payload: [number, number]) {}
+}
+
+export class ZoomEventAction implements Action {
+  readonly type = ActionTypes.ZoomEvent;
+  constructor(public payload: any) {}
+}
+
+export type Actions =
```

Comment:
Very picky but should the order of these follow the order of the declarations?

Reply:
I assume you're leaving this?

Reply:
I did this

Reply:
sorry, I wan't seeing your commits for some reason